Fix shortcuts in the Reflectometry GUI

raquelalvarezbanos opened the pull request:
Shortcuts to copy/cut/paste rows were not working in the Reflectometry GUI.

**To test:**

- Open `Interfaces` -> `Reflectometry` -> `ISIS Reflectometry`.
- Populate the table with some values, for example:

![reflectometry-table](https://cloud.githubusercontent.com/assets/8956985/23699546/73cd12ba-03e7-11e7-9d41-d9b3cc62a078.png)

- Check that you can copy/cut selected row(s) with Ctrl+c/Ctrl+x and paste them with Ctrl+v.

Fixes #19089.

Release notes: https://github.com/mantidproject/mantid/pull/19091/commits/b9ba05d55f73e81f3f64aa6d63de805caf4bb48d

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

gvardany commented:
When cutting the last existing row, there is a problem, first the little arrow on top left disappears, and it is not possible anymore to add a new row, and then when trying to paste, it crashes as in the figure:

![screen shot 2017-03-09 at 10 50 37](https://cloud.githubusercontent.com/assets/16865759/23745093/44e3a2c2-04b7-11e7-8cdb-9babe20f825f.png)

raquelalvarezbanos commented:
I've fixed the crash you were seeing and added a unit test. Note that at the moment, when there are no groups in the table, the only option to paste the row from the clipboard is to add a new group (6th icon in the toolbar), select the empty row and hit Ctrl+V. This behaviour may change in the future according to scientists feedback (this is the first release they are thoroughly testing the interface).

gvardany commented:
Does not crash anymore :shipit:
